server/models/message.model.js

```diff
@@ -53,7 +53,37 @@ module.exports = (sequelize, DataTypes) => {
             type: DataTypes.DATE,
             allowNull: true,
         },
-    });
+        isDeleted: {
+            type: DataTypes.BOOLEAN,
+            allowNull: false,
+            defaultValue: false
+        },
+        isArchived: {
+            type: DataTypes.BOOLEAN,
+            allowNull: false,
+            defaultValue: false
+        }
+      },
+      {
+        defaultScope: {
```

Comment:
Do we need to include `isArchived` here as well?

Reply:
Whatever you decide to do, let's document it clearly with comments and possibly in the README, since this will be important for how clients implement their message viewer.